lib/core/src/lib/viewer/components/viewer.component.spec.ts

```diff
@@ -525,6 +525,33 @@ describe('ViewerComponent', () => {
             });
         });
 
+        describe('Close Button', () => {
+
+            it('should show close button on left side when closeButtonPosition is left and allowGoBack is true', () => {
+                component.allowGoBack = true;
+                component.closeButtonPosition = 'left';
+                fixture.detectChanges();
+
+                expect(element.querySelector('.adf-left-close-button')).not.toBeNull();
+            });
+
+            it('should show close button on right side when closeButtonPosition is right and allowGoBack is true', () => {
+                component.allowGoBack = true;
+                component.closeButtonPosition = 'right';
+                fixture.detectChanges();
+
+                expect(element.querySelector('.adf-right-close-button')).not.toBeNull();
+            });
+
+            it('should hide close button allowGoBack is false', () => {
+                component.allowGoBack = false;
+                fixture.detectChanges();
+
+                expect(element.querySelector('.adf-right-close-button')).toBeNull();
+                expect(element.querySelector('.adf-left-close-button')).toBeNull();
```

Comment:
element.querySelector('.adf-left-close-button') is used multiple times, please move it to reusable function and reuse for each place